### PR TITLE
feat: factorsAI page and group call support

### PR DESCRIPTION
**Pull request by aashishmalik**
## Description of the change

>  factorsAI page and group call support

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/factorsai/transform.js [hunk 1]
```diff
@@ -41,6 +41,13 @@ function processTrack(message, factorsAIApiKey) {
   return buildResponse(requestJson, factorsAIApiKey);
 }
 
+// process Page Call
+function processPageAndGroup(message, factorsAIApiKey, eventType) {
+  const requestJson = eventType === EventType.PAGE ? constructPayload(message, mappingConfig[ConfigCategories.PAGE.name])
+      : constructPayload(message, mappingConfig[ConfigCategories.GROUP.name]);
+  return buildResponse(requestJson, factorsAIApiKey);
+}
```
Comment (koladilip) [inline src/v0/destinations/factorsai/transform.js:49]:
```suggestion
function processPageAndGroup(message, factorsAIApiKey, category) {
  const requestJson = constructPayload(message, mappingConfig[category]);
  return buildResponse(requestJson, factorsAIApiKey);
}
```

File: src/v0/destinations/factorsai/transform.js [hunk 2]
```diff
@@ -56,6 +63,10 @@ function process(event) {
       return processIdentify(message, factorsAIApiKey);
     case EventType.TRACK:
       return processTrack(message, factorsAIApiKey);
+    case EventType.PAGE:
+      return processPageAndGroup(message, factorsAIApiKey, EventType.PAGE);
+    case EventType.GROUP:
+      return processPageAndGroup(message, factorsAIApiKey, EventType.GROUP);
```
Comment (koladilip) [inline src/v0/destinations/factorsai/transform.js:69]: EventType.GROUP --> ConfigCategories.GROUP.name

File: src/v0/destinations/factorsai/transform.js [hunk 3]
```diff
@@ -56,6 +63,10 @@ function process(event) {
       return processIdentify(message, factorsAIApiKey);
     case EventType.TRACK:
       return processTrack(message, factorsAIApiKey);
+    case EventType.PAGE:
+      return processPageAndGroup(message, factorsAIApiKey, EventType.PAGE);
```
Comment (koladilip) [inline src/v0/destinations/factorsai/transform.js:67]: EventType.PAGE --> ConfigCategories.PAGE.name

File: src/v0/destinations/factorsai/transform.js (hunk 1 above)
Comment (aashishmalik) [reply src/v0/destinations/factorsai/transform.js]: updated

**Review by sanpj2292 — APPROVED** [review]

**Review by koladilip — APPROVED** [review]

**Pull request merged**